src/command_modules/azure-cli-interactive/azclishell/app.py

```diff
@@ -443,49 +434,45 @@ def example_repl(self, text, example, start_index, continue_flag):
         return cmd, continue_flag
 
     # pylint: disable=too-many-branches
-    def _special_cases(self, text, cmd, outside):
+    def _special_cases(self, cmd, outside):
         break_flag = False
         continue_flag = False
-        args = parse_quotes(text)
-        args_no_quotes = []
-        text_stripped = text.strip()
-        for arg in args:
-            args_no_quotes.append(arg.strip("/'").strip('/"'))
-
-        if text and len(text.split()) > 0 and text.split()[0].lower() == 'az':
-            telemetry.track_ssg('az', text)
-            cmd = ' '.join(text.split()[1:])
+        args = parse_quotes(cmd)
+        cmd_stripped = cmd.strip()
+        if cmd and len(cmd.split()) > 0 and cmd.split()[0].lower() == 'az':
```

Comment:
first of all `cmd.split` is invoked twice. second, all you need is the first word so you don't need to split everything (check out the split's manual). thrid, `if len(a_list) > 0` can be shorten to `if a_list`.

Reply:
fixed